cancellation.md

```diff
@@ -1879,6 +1934,89 @@ void grow(vector<int>& v){
+
+For each of the following events, each cancellation condition that is active at the start of the event is suspended for the duration of the event; after the event, such conditions are reactivated if their lifetime has not ended.
+- Stack unwinding
+- The evaluation of a cancellation condition
+- The execution of a noexcept function
+
+> [Here “noexcept” should be taken to mean “no early exit” — this will cause noexcept function to never be cancelled. This is preferred to introducing “nocancel” because the guarantee callers need doesn’t distinguish between exceptions and cancellation. This assumes that there is no use for guarantees of “no cancellation but maybe exceptions” or “no exceptions but maybe cancellation."  The name “noexcept” is unfortunate in this regard.]
+
+Just like we need exception_ptr to sneak exceptions across noexcept boundaries, we need a mechanism to sneak cancellation conditions across noexcept boundaries, but in the opposite direction. I think a library class will do the trick:
+
+class cancellation_test_functor
+    {
+    cancellation_test_functor() noexcept;
+        // Constructs a functor representing the cancellation conditions active at the time of construction.
+
+    cancellation_test_functor( const cancellation_test_functor& s ) noexcept;
+        // Constructs a functor representing the cancellation conditions represented by s.
+
+    bool operator() const noexcept;
+        // Evaluates the represented cancellation conditions, from eldest to youngest, stopping when one produces true.
+        // If the lifetime of any represented condition has ended, this function has undefined behavior.
+        // If any evaluated condition exits with an exception, it is treated as returning false.
+        // Returns true if any condition produces true; otherwise false.
+  };
+
+It is expected that this class can be implemented with just a pointer or two — it can just remember the point on the stack where it’s constructed (and use exception tables), or point to a range in a data structure tracking the stack of active conditions.
+
+This class doesn’t solve the synchronization issues of moving conditions from one thread to another. Since these conditions essentially use lambda-capture, that would be dangerous.  A mechanism to actively signal a cancellation request seems better than polling a condition across thread boundaries.  Perhaps an “I don’t need the answer to this” signal should be added to the future class.
```

Comment:
From email exchange:
 
   Lisa, " A mechanism to actively signal a cancellation request seems better than polling a condition across thread boundaries. "
    
    Kirk, "Yes, I agree. I think that the stop_source and stop_token are the mechanism for requesting cancellation. they only have to synchronize over state that they own - a bool and a list of callback functions, that return void and are noexcept, and they are designed to manage the lifetimes of both ends properly. Doing the same for the general case of a user supplied predicate that may be referencing a lot of arbitrary data appears to be intractable."